### PR TITLE
Fix draw.polygon segfault caused by 0d input

### LINKED ISSUE #4938: skimage.draw.polygon segmentation fault with np.nan
## Description
the function skimage.draw.polygon throws a segmentation fault instead of python exception when receiving np.nan

## Way to reproduce
```python
skimage.draw.polygon(np.nan, np.nan)
# Place the full code we need to recreate your issue here
# upload all necessary images to github too!
```

## Version information
```python
# Paste the output of the following python commands
from __future__ import print_function
import sys; print(sys.version)
import platform; print(platform.platform())
import skimage; print("scikit-image version: {}".format(skimage.__version__))
import numpy; print("numpy version: {}".format(numpy.__version__))
```

```python
# your output here

```

### PR DESCRIPTION
**Pull request by rfezzani**
## Description

Fixes #4938 

## Checklist

- [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- Gallery example in `./doc/examples` (new features only)
- Benchmark in `./benchmarks`, if your changes aren't covered by an
  existing benchmark
- Unit tests
- Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)

## For reviewers

- Check that the PR title is short, concise, and will make sense 1 year
  later.
- Check that new functions are imported in corresponding `__init__.py`.
- Check that new features, API changes, and deprecations are mentioned in
      `doc/release/release_dev.rst`.

### CODE REVIEW

File: skimage/draw/tests/test_draw.py [hunk 1]
```diff
@@ -214,6 +214,12 @@ def test_polygon_exceed():
     assert_array_equal(img, img_)
 
 
+def test_polygon_0d_input():
```
Comment (sciunto) [inline skimage/draw/tests/test_draw.py:217]: Please, add a comment with the issue number for future reference.

**Comment by sciunto** [comment]
@rfezzani Could you confirm that this shape is the only one that fails in the draw submodule?

**Review by sciunto — APPROVED** [review]

**Comment by rfezzani** [comment]
Thank you @sciunto.
No I can't confirm that `draw.polygon` is the only affected shape function, I simply searched if there is any `np.asanyarray` call in the `draw/_draw.pyx` file...

**Review by jni — APPROVED** [review]

**Comment by jni** [comment]
Thanks @rfezzani!

**Pull request merged**